### DISCUSSION TITLE
localhost refused to connect

### DISCUSSION THREAD

**Discussion by viviantic** [l3atbc-datadog/pushkin#114] [General] [answered]
Hello Pushkin Team! 

Sorry, here's another baby-level question from a newbie. I was building a basic site and an experiment following the Quickstart guidelines. When I tried to local-test it, I got "This site can't be reached. Localhost refused to connect." 

Here's what I have checked after I googled the issue: routers, network connections, browser's cache, and firewall settings. However, it still doesn't work. I am looking at the proxy settings now, but I am not sure how to adjust them for this site/experiment to work. Any suggestions would be greatly appreciated. Thank you.

**Comment by parkerrobbins** [comment, marked as answer]
Hi, @viviantic! Thank you for your question.

Can you tell me what operating system and web browser you are using?

**Reply by viviantic** [reply]
Hello @parkerrobbins! Thanks for getting back to me. 

The operating system I am using is macOS Catalina (10.15). I've tried google chrome, firefox and safari, but none of them worked.

**Reply by parkerrobbins** [reply]
@viviantic Sometimes web browsers will automatically try to redirect to `https`, but that won't work in this case. 

Can you confirm that&mdash;after you (1) run `pushkin start;` and (2) wait a few moments&mdash;`localhost` is resolved as `http://localhost` and not `https://localhost`? 

If it resolves to `https://localhost`, you could try manually editing it in the address bar to `http://localhost` (and press ENTER) and see if that allows you to access it.

**Reply by viviantic** [reply]
Hello @parkerrobbins! Thanks for your answer. I think the problem might be that I haven't been able to run pushkin start successfully. This is what I got after I ran pushkin start: 

"Pulling vocab_worker (vocab_worker:)...
The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.
Continue with the new image? [yN]" 

I've tried both [yN] followed by local testing, but none of them got connected. By the way, the address I tried was http://localhost.

**Reply by parkerrobbins** [reply]
Hi, @viviantic! Thanks for your response!

A couple of ideas.

1. Check your Pushkin version by running `pushkin --version`. If it is less than `1.1.5`, you can upgrade Pushkin by running `yarn global upgrade pushkin-cli`. When you run `pushkin --version` again, it should be at `1.1.5`.

2. Try running `pushkin kill` and trying again. If that doesn't work, you can run `pushkin armageddon` and try again. (Make sure you run both `pushkin prep` and `pushkin start;` after this.)

**Reply by viviantic** [reply]
Hello @parkerrobbins! Thanks for the suggestions. The version is 1.1.5, and I ran all that you suggested. Unfortunately, I still got the same result: 

"Pulling vocab_worker (vocab_worker:)...
The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.

Continue with the new image? [yN]"

**Reply by parkerrobbins** [reply]
Thanks for your response. I'm not totally sure what's going on, so I will defer to @jkhartshorne, who will probably have more ideas.

**Reply by jkhartshorne** [reply]
@viviantic can you please post the the following:

1. The names of the folders found in `experiments/`

2. The contents of `pushkin/front-end/src/experiments.js`

3. The contents of `pushkin/docker-compose.dev.yml`

**Reply by viviantic** [reply]
Hello @jkhartshorne! 

1. The names of the folders found in experiments/
mind, vocab, and whichenglish

2. the contents of pushkin/front-end/src/experiments.js

import undefined from 'undefined';
import mind_web from 'mind_web';
import vocab_web from 'vocab_web';
import whichenglish_web from 'whichenglish_web';
export default [
undefined
,{ fullName: `mind`,shortName: 'mind', 
    module: mind_web, logo: 'logo512.png', 
    tagline: 'Be a citizen scientist! Try this quiz.', 
    duration: '', 
    text: 'Enter your experiment description here.' }
,{ fullName: `vocab`,shortName: 'vocab', 
    module: vocab_web, logo: 'logo512.png', 
    tagline: 'Be a citizen scientist! Try this quiz.', 
    duration: '', 
    text: 'Enter your experiment description here.' }
,{ fullName: `whichenglish`,shortName: 'whichenglish', 
    module: whichenglish_web, logo: 'logo512.png', 
    tagline: 'Be a citizen scientist! Try this quiz.', 
    duration: '', 
    text: 'Enter your experiment description here.' }
];

3. The contents of pushkin/docker-compose.dev.yml

version: '3.1'
volumes:
  message_queue_volume: null
  test_db_volume: null
services:
  api:
    build: ./api
    environment:
      - 'AMQP_ADDRESS=amqp://message-queue:5672'
      - PORT=3000
    expose:
      - '3000'
    links:
      - message-queue
  message-queue:
    image: 'rabbitmq:3.6-management'
    expose:
      - 5672
      - 15672
    ports:
      - 15672
    environment:
      CONFD_ARGS: '--interval 5'
      RABBITMQ_CLUSTER_PARTITION_HANDLING: autoheal
      RABBITMQ_NET_TICKTIME: '60'
      RABBITMQ_ERLANG_COOKIE: message-queue-cookie
      RABBITMQ_NODENAME: rabbitmqnode@localhost
    volumes:
      - 'message_queue_volume:/var/lib/rabbitmq'
  server:
    build: ./front-end
    environment:
      API_PORT: '3000'
    ports:
      - '80:80'
      - '433:433'
    links:
      - api
  test_db:
    image: 'postgres:11'
    environment:
      POSTGRES_PASSWORD: example
      POSTGRES_DB: test_db
    ports:
      - '5432:5432'
    volumes:
      - 'test_db_volume:/var/lib/postgresql/data'
    healthcheck:
      test:
        - CMD-SHELL
        - pg_isready -U postgres
      interval: 10s
      timeout: 5s
      retries: 5
  vocab_worker:
    image: vocab_worker
    links:
      - message-queue
      - test_db
    environment:
      - 'AMQP_ADDRESS=amqp://message-queue:5672'
      - DB_USER=postgres
      - DB_PASS=
      - DB_URL=test_db
      - DB_NAME=test_db
    labels:
      isPushkinWorker: true
  mind_worker:
    image: mind_worker
    links:
      - message-queue
      - test_db
    environment:
      - 'AMQP_ADDRESS=amqp://message-queue:5672'
      - DB_USER=postgres
      - DB_PASS=
      - DB_URL=test_db
      - DB_NAME=test_db
    labels:
      isPushkinWorker: true
  whichenglish_worker:
    image: whichenglish_worker
    links:
      - message-queue
      - test_db
    environment:
      - 'AMQP_ADDRESS=amqp://message-queue:5672'
      - DB_USER=postgres
      - DB_PASS=
      - DB_URL=test_db
      - DB_NAME=test_db
    labels:
      isPushkinWorker: true

**Reply by viviantic** [reply]
Hello @jkhartshorne @parkerrobbins! 

I finally solved the problem. It seems that I haven't been able to run pushkin start successfully. It finally works after I killed everything, cleaned all the cookies and browser history, re-installed pushkin, and created a new site basically all over again. I don't know why, but It's working this way for me. 

Thank you so much for your help and for being so patient with this issue.

**Reply by jkhartshorne** [reply]
There was something wrong with your `experiments.js` file. You can see in that first line that `import undefined` command. That won't run. I'm not sure how that happened. Glad it's working now.

**Comment by restmegan** [comment]
Hi @viviantic ! I just wanted to check in to see how Pushkin has been going for you. The Pushkin team is online today answering questions so if there's anything you need help with please let us know!
